### PR TITLE
Compile on MacOS 

### LINKED ISSUE #26: Compiling with Boost thread on Mac
When compiling the dso_dataset binary on Mac, I encountered the following error:
```
ld: library not found for -lboost_thread
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/dso_dataset] Error 1
make[1]: *** [CMakeFiles/dso_dataset.dir/all] Error 2
make: *** [all] Error 2
```

### PR DESCRIPTION
**Pull request by AlexandrGraschenkov**
fix #26

Also I have problem with code compile. Fixed by changing type of `connectivityMap` to:

    std::map<uint64_t,Eigen::Vector2i> connectivityMap;

### CODE REVIEW

**Comment by NikolausDemmel** [comment]
Thanks for the PR.

I guess we could also change the type of ConnectivityMap for all platforms, no?

**Comment by AlexandrGraschenkov** [comment]
@NikolausDemmel updated. Not sure is it will compile normally on Linux.

**Comment by NikolausDemmel** [comment]
Thanks for the update.

I think removing the a aligning allocator is problematic. It doesn't result in a compile time error, but it might result in runtime alignment errors (which can sometimes happen and sometimes not, which makes them hard to track down).

What is the compilation error with the ConnectivityMap as is?

**Comment by NikolausDemmel** [comment]
Also, which version of OS X and which compiler?

**Comment by AlexandrGraschenkov** [comment]
Error like so:

> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:820:5: error: 
>       static_assert failed "Allocator::value_type must be same type as
>       value_type"
>     static_assert((is_same<typename allocator_type::value_type, value_type>::value),
>     ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /Users/alex/Documents/roadar/dso/src/OptimizationBackend/EnergyFunctional.h:121:6: note: 
>       in instantiation of template class 'std::__1::map<unsigned long long,
>       Eigen::Matrix<int, 2, 1, 0, 2, 1>, std::__1::less<unsigned long long>,
>       Eigen::aligned_allocator<std::__1::pair<unsigned long long,
>       Eigen::Matrix<int, 2, 1, 0, 2, 1> > > >' requested here
>           > connectivityMap;

AppleClang 9.0.0.9000037, Mac OS Siera 10.12.6

**Comment by NikolausDemmel** [comment]
Hm... still curious about why more recent clang chokes on the allocator. But I'll merge for now. Thanks for iterating on this!

File: CMakeLists.txt [hunk 1]
```diff
@@ -115,11 +115,17 @@ add_library(dso ${dso_SOURCE_FILES} ${dso_opencv_SOURCE_FILES} ${dso_pangolin_SO
 #set_property( TARGET dso APPEND_STRING PROPERTY COMPILE_FLAGS -Wall )
 
 
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin") # OSX
+    set(BOOST_THREAD_LIBRARY boost_thread-mt)
+else()
+    set(BOOST_THREAD_LIBRARY boost_thread)
+endif()
+
```
Comment (NikolausDemmel) [inline CMakeLists.txt]: P.S.: Just for the record, I noticed that cmake provides this as `Boost_THREAD_LIBRARY`, so we could have just used that... I will correct this at some point.

**Pull request merged**